### PR TITLE
fix: Fixed event source mapping filter criteria

### LINKED ISSUE #271: Event source filter criteria does not work
## Description

Following the example [here](https://github.com/terraform-aws-modules/terraform-aws-lambda/tree/master/examples/event-source-mapping) you get an empty filter in the filter_criteria of the resource `aws_lambda_event_source_mapping`.

```
Terraform will perform the following actions:

  # module.lambda_function.aws_lambda_event_source_mapping.this["dynamodb"] will be updated in-place
  ~ resource "aws_lambda_event_source_mapping" "this" {
        id                                 = "a4f49422-beff-4a9d-b7c4-8bd94a09bf2c"
        # (20 unchanged attributes hidden)


      + filter_criteria {
          + filter {}
        }
        # (1 unchanged block hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
╷
│ Warning: Argument is deprecated
│
│   with module.lambda_function.aws_s3_bucket_object.lambda_package,
│   on .terraform/modules/lambda_function/main.tf line 124, in resource "aws_s3_bucket_object" "lambda_package":
│  124:   bucket        = var.s3_bucket
│
│ Use the aws_s3_object resource instead
│
│ (and one more similar warning elsewhere)
╵

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

module.lambda_function.aws_lambda_event_source_mapping.this["dynamodb"]: Modifying... [id=a4f49422-beff-4a9d-b7c4-8bd94a09bf2c]
╷
│ Error: error updating Lambda Event Source Mapping (a4f49422-beff-4a9d-b7c4-8bd94a09bf2c): InvalidParameterValueException: Invalid filter pattern definition.
│ {
│   RespMetadata: {
│     StatusCode: 400,
│     RequestID: "6b5a928a-e4ef-42c4-89c0-59a4d92503c4"
│   },
│   Message_: "Invalid filter pattern definition.",
│   Type: "User"
│ }
│
│   with module.lambda_function.aws_lambda_event_source_mapping.this["dynamodb"],
│   on .terraform/modules/lambda_function/main.tf line 223, in resource "aws_lambda_event_source_mapping" "this":
│  223: resource "aws_lambda_event_source_mapping" "this" {
```
## Versions
```
$ terraform -version
Terraform v1.1.6
on linux_amd64
+ provider registry.terraform.io/hashicorp/aws v4.2.0
+ provider registry.terraform.io/hashicorp/external v2.2.0
+ provider registry.terraform.io/hashicorp/local v2.1.0
+ provider registry.terraform.io/hashicorp/null v3.1.0
```

Module version: 2.34.0

## Reproduction

```
event_source_mapping = {
    dynamodb = {
      event_source_arn           = module.services.dynamodb_table_stream_arn
      starting_position          = "LATEST"
      destination_arn_on_failure = aws_sqs_queue.failure.arn
      filter_criteria = {
        pattern = jsonencode({
          eventName = ["INSERT", "MODIFY"]
        })
      }
    }
  }
```
## Expected behavior
A filter criteria like this:
```
    + filter_criteria {
          + filter {
              + pattern = jsonencode(
                    {
                      + eventName = [
                          + "INSERT",
                          + "MODIFY",
                        ]
                    }
                )
            }
        }
```

## Actual behavior
An empty filter criteria:
```

      + filter_criteria {
          + filter {}
        }
```

### PR DESCRIPTION
**Pull request by sindrig**
This closes #271

## Description
See #271

## How Has This Been Tested?
- [X] I have tested and validated these changes using one or more of the provided `examples/*` projects

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -262,7 +262,7 @@ resource "aws_lambda_event_source_mapping" "this" {
 
     content {
       filter {
-        pattern = lookup(filter_criteria, "pattern", null)
+        pattern = lookup(each.value["filter_criteria"], "pattern", null)
```
Comment (bryantbiggs) [inline main.tf:265]: I think instead we should change line 261 to

```hcl
for_each = try(each.value.filter_criteria, [])
```

which should make 265 work as intended without changes (I think, haven't tried myself yet)

Comment (antonbabenko) [reply main.tf:265]: @bryantbiggs This module is still full of `lookup()` expressions, so using `try()` is not so critical here.

Comment (sindrig) [reply main.tf:265]: I wanted to follow what's done elsewhere in this module. I can change it if you want though.

**Comment by antonbabenko** [comment]
Thanks @sindrig !

**Comment by antonbabenko** [comment]
This PR is included in version 2.34.1 :tada:

**Pull request merged**